backend/src/services/user/login.ts

```diff
@@ -103,22 +106,38 @@ const getToken = async (
   }
 
   if (user && (await comparePassword(basicAuthUser.pass, user.hash))) {
-    return await generateToken(user);
+    return await generateToken(user, untilMidnight);
   }
   return null;
 };
 
-const generateToken = async (user: User) => {
+export const calculateExpirationTimeStamp = (
+  issuedAtTime: Date,
+  expirationTimeInMs: number,
+  untilMidnight?: boolean,
+): number => {
+  untilMidnight = untilMidnight === true;
+
+  if (untilMidnight) {
+    const nextMidnight = new Date(issuedAtTime);
+    nextMidnight.setHours(23, 59, 59, 0);
+    return nextMidnight.getTime();
+  }
+  return issuedAtTime.getTime() + expirationTimeInMs;
+};
+
+const generateToken = async (user: User, untilMidnight?: boolean) => {
   await AppDataSource.getRepository(Token).delete({
     user: { id: user.id },
   });
   const token = new Token();
   token.jti = randomUUID();
   token.iat = new Date();
-  let expirationTime = token.iat.getTime() + config.jwt.expirationTimeInMs;
-  if (user.role == UserRole.USER) {
-    expirationTime += 3 * config.jwt.expirationTimeInMs;
-  }
+  const expirationTime = calculateExpirationTimeStamp(
+    token.iat,
+    config.jwt.expirationTimeInMs * (user.role == UserRole.USER ? 3 : 1),
```

Comment:
Vorher war es das Vierfache (1+3), Absicht?

Reply:
das war ein Versehen ... aber vermutlich auch nicht dramatisch. Dann ist die Auto-Logout-Zeit nur 3 statt 4 Stunden.

Ein Token-Refresh wäre auch meine angestrebte Lösung. Aber evtl. trotzdem in Kombination mit Auto-Logout?

In #35 sind bereits ein paar mögliche Varianten notiert.

Reply:
Ja, denke Auto-Login bei Inaktivität. Aber ich hab da auch nicht den Einblick, wie man das "heute" so macht mit den JWTs ^^